tensorflow/compiler/xla/service/gpu/gpu_executable.cc

```diff
@@ -425,7 +433,29 @@ StatusOr<ExecutionOutput> GpuExecutable::ExecuteAsyncOnStream(
     const ServiceExecutableRunOptions* run_options,
     std::vector<ExecutionInput> arguments,
     HloExecutionProfile* hlo_execution_profile) {
-  XLA_SCOPED_LOGGING_TIMER(absl::StrCat("GpuExecutable::ExecuteAsyncOnStream(",
+  return ExecuteAsyncOnStreamImpl(run_options, /*arguments_buffer*/nullptr,
+                                  &arguments, hlo_execution_profile);
+}
+
+StatusOr<ScopedShapedBuffer> GpuExecutable::ExecuteAsyncOnStream(
+      const ServiceExecutableRunOptions* run_options,
+      absl::Span<const ShapedBuffer* const> arguments,
+      HloExecutionProfile* hlo_execution_profile) {
+  TF_ASSIGN_OR_RETURN(ExecutionOutput out,
+                      ExecuteAsyncOnStreamImpl(run_options, &arguments,
+                                               /*arguments_input*/nullptr,
+                                               hlo_execution_profile));
+  return out.ConsumeResult();
+
+}
+
+StatusOr<ExecutionOutput> GpuExecutable::ExecuteAsyncOnStreamImpl(
+      const ServiceExecutableRunOptions* run_options,
+      absl::Span<const ShapedBuffer* const>* arguments_buffer,
+      std::vector<ExecutionInput>* arguments_inputs,
+      HloExecutionProfile* hlo_execution_profile) {
+  CHECK_EQ((arguments_buffer == nullptr) + (arguments_inputs == nullptr), 1);
```

Comment:
Is it `CHECK(!arguments_buffer ^ !arguments_inputs)`?

Reply:
Right. Changed.